gcc/rust/ast/rust-ast-full-test.cc

```diff
@@ -4050,6 +4052,88 @@ Module::add_crate_name (std::vector<std::string> &names) const
     item->add_crate_name (names);
 }
 
+static bool
+check_for_file (std::string dir_name, std::string file_name)
+{
+  DIR *dir = opendir (dir_name.c_str ());
+  if (!dir)
+    return false;
+
+  struct dirent *entry;
+
+  while ((entry = readdir (dir)) != NULL)
+    {
+      if (std::string (entry->d_name) == file_name)
+	return true;
+    }
+
+  closedir (dir);
+
+  return false;
+}
+
+// FIXME: This function should also check if the module has a `path` outer
+// attribute and fetch the path from here in that case, i.e:
+// ```
+// #[path="<dir>/<subdir>/<file>.rs"]
+// mod <mod_name>;
+// ```
+std::string
+Module::get_filename ()
+{
+// TODO: Not windows compliant, at least not always. We need to have some
+// sort of #ifdef but where should it be put? Since this could also be used
+// elsewhere in the compiler
+#define SEPARATOR "/"
```

Comment:
I think a static constant might be more C++'y

Reply:
Done (I think? Not too sure about `constexpr static const`) in e484f42